lib/database/sql/000020_dial_failure.up.sql

```diff
@@ -0,0 +1,34 @@
+CREATE TABLE IF NOT EXISTS dialFailure (
+    id SERIAL PRIMARY KEY,
+    recordId INTEGER NOT NULL,
+    errorType INTEGER NOT NULL,
+    errorMsg TEXT,
+    protocols TEXT NOT NULL,
+    timestamp INTEGER NOT NULL,
+    CONSTRAINT dialFailure_unique UNIQUE (recordId, errorType, protocols, timestamp)
+);
+
+ALTER TABLE dialFailure ADD CONSTRAINT fk_dialFailure_telemetryRecord
```

Comment:
Shouldn't there be a foreign key constraint for `dial_error_types` as well? Just to make sure we don't insert non-existant error types?